src/test/java/uk/gov/justice/digital/delius/service/ConvictionServiceTest.java

```diff
@@ -250,11 +250,33 @@ public void convictionReturnedWhenSingleActiveConvictionMatchedAmoungstDuplicate
         public void emptyReturnedWhenNoConvictionsMatchedForOffenderIdAndPrisonBookingNumber() throws DuplicateConvictionsForBookingNumberException {
             when(convictionRepository.findByOffenderIdAndPrisonBookingNumber(99L, "A12345")).thenReturn(ImmutableList.of());
 
-            Optional<Event> maybeConviction = convictionService.getSingleActiveConvictionIdByOffenderIdAndPrisonBookingNumber(99L, "A12345");
+            Optional<Event> maybeConviction = convictionService.getSingleActiveConvictionByOffenderIdAndPrisonBookingNumber(99L, "A12345");
 
             assertThat(maybeConviction).isNotPresent();
         }
 
+
+        @Test
+        public void convictionIdReturnedWhenSingleActiveConvictionMatchedAmongstDuplicatesForOffenderIdPrisonBookingNumber() throws DuplicateConvictionsForBookingNumberException {
+            when(convictionRepository.findByOffenderIdAndPrisonBookingNumber(99L, "A12345")).thenReturn(ImmutableList.of(
+                    aEvent()
+                            .toBuilder()
+                            .eventId(998L)
+                            .activeFlag(0L)
+                            .build(),
+                    aEvent()
+                            .toBuilder()
+                            .eventId(999L)
+                            .activeFlag(1L)
+                            .build()
+            ));
+
+            Optional<Long> maybeConviction = convictionService.getSingleActiveConvictionIdByOffenderIdAndPrisonBookingNumber(99L, "A12345");
+
+            assertThat(maybeConviction.orElseThrow())
```

Comment:
Just wondering what the `orElseThrow` brings to the table?

Reply:
I suppose it is the same as writing `assertThat(maybeConviction).get().isEqualTo(999L)`  notice if I did `assertThat(maybeConviction.get()).isEqualTo(999L)` IntelliJ would warn me. So I suppose it is saying "I really expect to have a value here, but if don't throw an exception and fail"